src/xcode/ENA/ENA/Source/Scenes/Home/HomeViewController.swift

```diff
@@ -403,21 +415,35 @@ final class HomeViewController: UIViewController {
 
 extension HomeViewController {
 
-	func updateTestResultFor(_ cell: HomeTestResultCell, with configurator: HomeTestResultCellConfigurator) {
+	func showTestResultScreen() {
+		showExposureSubmission(with: testResult)
+	}
+
+	func updateTestResultState() {
+		self.homeInteractor.showTestResultCell()
+		fetchTestResult()
+	}
+
+	private func fetchTestResult() {
+		DispatchQueue.global(qos: .userInteractive).async {
+			self.fetchTestResultHelper()
+		}
+	}
+
+	private func fetchTestResultHelper() {
+		// Make sure we are able to fetch.
+		guard store.registrationToken != nil else { return }
 		self.exposureSubmissionService?.getTestResult { result in
 			switch result {
 			case .failure(let error):
-				appLogger.log(message: "Could not update test state: \(error)", file: #file, line: #line, function: #function)
+				appLogger.log(message: "Error while fetching result: \(error)", file: #file, line: #line, function: #function)
 			case .success(let result):
 				self.testResult = result
-				configurator.configure(cell: cell)
+				self.homeInteractor.reloadTestResult(with: result)
 			}
 		}
 	}
```

Comment:
business logic should be in interactor or in another class, not in view controller

Reply:
i moved that to the interactor